votor/src/certificate_pool.rs

```diff
@@ -668,20 +636,18 @@ impl CertificatePool {
     /// Cleanup any old slots from the certificate pool
     pub fn handle_new_root(&mut self, bank: Arc<Bank>) {
```

Comment:
who is calling this and does it need to pass in the bank.

Reply:
this should not be needed at all: pool itself determines finalization, finalization itself is enough to trigger cleanup within pool

Reply:
the only use for bank here was to support epoch stakes. If we do not have epoch stakes then we can immediately pass the finalized slot here in the caller (and avoid having to take the lock in the caller).

Just a side note, RootBankCache was removed in master, it'd be a good idea to cherry-pick and update to the new SharableRootBank at some point. https://github.com/anza-xyz/agave/pull/7300

Reply:
Indeed... I plan on making these changes in steps so as to keep the PR small and self contained.